league.js

```diff
@@ -0,0 +1,296 @@
+            if (details.opponent) {
+                lichess.getPlayerRating(details.opponent).then(function(rating) {
+                    details['rating'] = rating;
+                    deferred.resolve(details);
+                }, function(error) {
+                    console.error(JSON.stringify(error));
+                    deferred.resolve(details);
+                });
+            } else {
+                deferred.resolve({});
+            }
+            return deferred.promise;
+        });
+    },
+    //--------------------------------------------------------------------------
+    // Formats the pairing result for this league
+    //--------------------------------------------------------------------------
+    'formatPairingDetails': function(requestingPlayer, details) {
+        function getRatingString(rating){
+            return ( rating ? " (" + rating + ")" : "" );
+        }
+        var self = this;
+        return Q.fcall(function() {
+            var localTime;
+            if (details.date) {
+               localTime = requestingPlayer.localTime(details.date);
+            }
+            var schedule_phrase = "";
+            var played_phrase = "";
+
+            if (!localTime || !localTime.isValid()) {
+                played_phrase = "will play as";
+                schedule_phrase = ". The game is unscheduled.";
+            } else if (moment.utc().isAfter(localTime)) {
+                // If the match took place in the past, display the date instead of the day
+                played_phrase = "played as";
+                schedule_phrase = " on {localDateTimeString}.".format({
+                    localDateTimeString: localTime.format("MM/DD [at] HH:mm")
+                });
+            } else {
+                played_phrase = "will play as";
+                schedule_phrase = " on {localDateTimeString} which is in {timeUntil}".format({
+                    localDateTimeString: localTime.format("MM/DD [at] HH:mm"),
+                    timeUntil: localTime.fromNow(true)
+                })
+            }
+
+            // Otherwise display the time until the match
+            return "[{name}]: {details.player} {played_phrase} {details.color} against {details.opponent}{rating}{schedule_phrase}".format({
+                name: self.options.name,
+                details: details,
+                played_phrase: played_phrase,
+                schedule_phrase: schedule_phrase,
+                rating: getRatingString(details.rating),
+            });
+        });
+    },
+};
+
+function League(options) {
+    this.options = {};
+    _.extend(this.options, LEAGUE_DEFAULTS, options || {});
+    _.extend(this, league_attributes);
+};
+
+function getAllLeagues(config) {
+    var leagues = [];
+    var all_league_configs = config['leagues'] || {};
+    _.each(_.keys(all_league_configs), function(key) {
+        leagues.push(getLeague(key, config));
+    });
+    return leagues;
+}
+
```

Comment:
@endrawes0 Note, I tried your version but the this was different for whatever context you called it in. At least that was my assessment when this function was creating the same league twice from two different contexts.  This has a similar effect (the _league_cache is not in the global scope), but is more explicit about the scope and ensures that the _league_cache is effectively static (one copy per process).